app/models/requirement.rb

```diff
@@ -94,30 +94,31 @@ def number_unit
     input_options["number_unit"]
   end
 
+  def key
+    "#{requirement_block.key}|#{requirement_code}"
+  end
+
   def to_form_json
-    json = {
-      id: id,
-      key: label.parameterize.underscore.camelize(:lower),
-      type: input_type,
-      input: true,
-      label: label,
-      widget: {
-        type: "input",
-      },
-    }.merge!(formio_type_options)
+    json = { id: id, key: key, type: input_type, input: true, label: label, widget: { type: "input" } }.merge!(
+      formio_type_options,
+    )
 
     if input_type_select? || input_type_multi_option_select?
       json.merge!({ data: { values: input_options["value_options"] } })
     end
 
+    if input_options["computed_compliance"].present?
+      json.merge!({ computedCompliance: input_options["computed_compliance"] })
+    end
+
     json
   end
 
   private
 
   # requirement codes should not be auto generated during seeding.  Use uuid if not provided
   def set_requirement_code
-    self.requirement_code ||= SecureRandom.uuid
+    self.requirement_code ||= label.present? ? label.parameterize.underscore.camelize(:lower) : SecureRandom.uuid
```

Comment:
For this to work with the key method, we have to write a validation to ensure that the label is not the same as another requirement's label in the same requirement block

Reply:
@taw27 - doe sit need to be unique? I guess the code needs to be unique per requirement block, not against all rewquirements.

Reply:
Never mind I misunderstood something